ChromeExtension/js/apiKeyViewModel.js

```diff
@@ -1,75 +1,64 @@
-function apiKeyViewModel() {
-
-
-    var self = this;
-    self.apiKey = ko.observable('');
-    self.apiSecret = ko.observable('');
-    self.showBackLink = ko.observable(false);
-    self.showHelpLink = ko.observable(true);
-    self.newInstall = ko.observable(true);
-    self.UrlApiKeys = ko.observable('');
-
-
-    self.apiKey.subscribe(function (newValue) {
-        localStorage.setItem("apiKey", newValue);
-    });
-
-    self.apiSecret.subscribe(function (newValue) {
-        localStorage.setItem("apiSecret", newValue);
-    });
-
-
-    self.loadKey = function () {
-        var ak = localStorage["apiKey"];
-        if (typeof ak !== 'undefined') {
-            self.apiKey(ak);
-        }
-        var asa = localStorage["apiSecret"];
-        if (typeof asa !== 'undefined') {
-            self.apiSecret(asa);
-        }
+var apiKeyViewModel = {
+    apiKey: ko.observable(''),
+    apiSecret: ko.observable(''),
+    showBackLink: ko.observable(false),
+    showHelpLink: ko.observable(true),
+    newInstall: ko.observable(true),
+    UrlApiKeys: ko.observable('')
+};
+
+apiKeyViewModel.apiKey.subscribe(function (newValue) {
+    chrome.storage.local.set({"apiKey": newValue});
+});
 
-    }
+apiKeyViewModel.apiSecret.subscribe(function (newValue) {
+    chrome.storage.local.set({"apiSecret": newValue});
+});
 
-    if (localStorageHasValue("defaultGroup")) {
-        self.showBackLink(true);
-        self.showHelpLink(false);
-        self.newInstall(false);
-        self.UrlApiKeys = ('https://my.geni.us/tools#api-section');
+chrome.storage.local.get(["defaultGroup"]).then((group) => {
```

Comment:
there's changes to ALL expressions that use localStorage to using `chrome.storage.local` due to this:
https://developer.chrome.com/docs/extensions/develop/migrate/to-service-workers#convert-localstorage

> The web platform's [Storage](https://developer.mozilla.org/docs/Web/API/Storage) interface (accessible from window.localStorage) cannot be used in a service worker. To address this do one of two things. First, you can replace it with calls to another storage mechanism. The [chrome.storage.local](https://developer.chrome.com/docs/extensions/reference/storage#property-local) namespace will serve most use cases, but [other options](https://developer.chrome.com/docs/extensions/mv3/service_workers/service-worker-lifecycle#persist-data) are available.